app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java

```diff
@@ -0,0 +1,35 @@
+package com.teamagam.gimelgimel.app.mainActivity.viewmodel;
+
+import com.teamagam.gimelgimel.app.injectors.scopes.PerActivity;
+
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
+/**
+ * Created by Admin on 13/12/2016.
+ */
+
+@PerActivity
+public class PanelViewModel {
+
+    private String[] mTitles;
+    private int mNumOfTitles;
+
+
+    @Inject
+    public PanelViewModel() {}
+
+    public void setArguments (String[] titles, int numOfTitles) {
```

Comment:
There's no point in asking for an array and it's length.
One can simply ask the array for it's length when required.